opentracing-apache-httpclient/src/main/java/io/opentracing/contrib/apache/http/client/ApacheClientSpanDecorator.java

```diff
@@ -56,12 +56,20 @@
 
         @Override
         public void onRequest(HttpRequestWrapper request, HttpContext httpContext, Span span) {
+            URI uri = request.getURI();
+            HttpHost target = request.getTarget();
+
             Tags.HTTP_METHOD.set(span, request.getRequestLine().getMethod());
 
-            URI uri = request.getURI();
-            Tags.HTTP_URL.set(span, request.getRequestLine().getUri());
-            Tags.PEER_PORT.set(span, uri.getPort() == -1 ? 80 : uri.getPort());
-            Tags.PEER_HOSTNAME.set(span, uri.getHost());
+            if (uri != null) {
+                Tags.HTTP_URL.set(span, request.getRequestLine().getUri());
```

Comment:
Why not to use directly the `uri`variable ?

Reply:
This can be done.  I kept it the same since it wasn't breaking but I agree this makes sense.  Will push the update.